### PR TITLE
fix(quickadapter): recheck weight support after pipeline

### LINKED ISSUE #93: quickadapter: label-weighting support not rechecked after `feature_pipeline.fit_transform`
## Context

PR #90 routes train-side label-weight composition through `_compose_train_weights_with_support`, enforcing `support_policy` at compose time. `feature_pipeline.fit_transform` (`_train_common` in `QuickAdapterRegressorV3.py`) later removes training rows and renormalizes the weight column on the surviving rows.

## Defect

The composed weights satisfy the support contract at compose time, but the post-pipeline weights -- the ones actually passed to `fit` -- can fall below `min_pivot_equivalent_count`, `min_positive_label_weight_fraction`, or `min_effective_sample_size` without triggering `support_policy`. `raise` does not raise; `fallback` does not warn.

## Proposal

Thread the support check through the pipeline pass: re-evaluate `summarize_label_weight_support` on the post-`fit_transform` weights and apply `_apply_support_policy` on threshold violation. Touches `_train_common` and the feature-pipeline integration in `QuickAdapterRegressorV3.py`.

## Reference

Follow-up from PR #80 review (deferred during PR #90 implementation).

### PR DESCRIPTION
**Pull request by jerome-benoit**
## Summary

- carry raw base and label-weight components through FreqAI feature-pipeline row filtering only when label weighting is active
- reuse the existing train-support enforcement on the final weights for selection training and final refit
- restore the feature-pipeline label schema and fail closed if `fit_transform` or eval `transform` drops or reshapes `sample_weight`

## Root cause

QuickAdapter enforced label-weight support after splitting and causal guards. `feature_pipeline.fit_transform` could then remove training rows and renormalize the surviving weights below the configured pivot-count, positive-fraction, or effective-sample-size thresholds without invoking `support_policy`.

DataSieve resets DataFrame indexes during `fit_transform`, so returned indexes cannot recover the surviving raw weight components. Its `y` channel is the existing row-aligned API: FreqAI transforms apply the same outlier mask to `X`, `y`, and `sample_weight`, while `SGDOneClassSVM.fit` ignores `y`. Temporary object-keyed columns carry the two raw components, are removed immediately afterward, and the original public label schema is restored before validation/test transforms.

The shared post-call shape check is a hard pipeline-contract invariant. It intentionally does not route `None` or malformed weights through statistical `support_policy` fallback.

## Scope and compatibility

- backtest and live retraining share the same `train()` path; selection training, HPO, holdout evaluation, and final refit preserve their existing weight routing
- the no-label-weight path passes the original labels directly to DataSieve
- eval weights continue to bypass the training-only support policy, but malformed pipeline output now fails closed
- direct estimator calls use each library's documented weight API; enabling sklearn metadata routing does not alter these direct-consumer calls
- no tunable, canonical default, public Freqtrade API, or model-reset requirement changes
- no functional overlap with #148 or #151

## Validation

Focused harnesses are external under `/tmp`; no tests are committed.

- `/tmp/qa150_owner_main_repro.py`: reproduced #93 on current `upstream/main` `b19cf6d9235dda3f7a3d4d75363de257a0c2ebbf`; a post-pipeline all-zero support set was silently uniformized without a support recheck
- `/tmp/qa150_owner_pipeline_oracle.py`: Freqtrade `develop` source `308a048abfaffdd85aaaa1f27995dfce77bf2e04`, DataSieve `0.1.9`, and scikit-learn `1.9.0`; verified post-filter `raise`/`fallback`, row reordering and index reset, schema restoration, unchanged no-label behavior, NaN/Inf/empty handling, train/refit hard failure for `None`/scalar/2-D/wrong-length weights, and validation/holdout hard failure when weights are lost
- the same pipeline oracle ran a real `SVMOutlierExtractor` with sklearn metadata routing enabled; 4/80 rows were removed while feature/label/weight alignment and the original label schema were preserved
- `/tmp/qa150_owner_regressor_oracle.py`: real weighted and uniform fits through XGBoost `3.2.0`, LightGBM `4.7.0`, HistGradientBoostingRegressor from scikit-learn `1.9.0`, NGBoost `0.5.11`, and CatBoost `1.2.10`; all weighted fits were finite and observably different
- the regressor oracle also verified exact HPO train/eval weight routing, weighted validation RMSE, weighted holdout RMSE, and the final-refit context/weights
- `ruff check` and `ruff format --check` on the model and supporting QuickAdapter module
- Python 3.14 `compileall`
- `git diff --check`

Fixes #93

### CODE REVIEW

File: quickadapter/user_data/freqaimodels/QuickAdapterRegressorV3.py [hunk 1]
```diff
@@ -2468,34 +2505,68 @@ def _add_refit_data(
...
         )
+        if weight_inputs.label is not None:
+            post_pipeline_base_weights = pipeline_labels.pop(
+                base_weight_column
+            ).to_numpy(dtype=float)
+            post_pipeline_label_weights = pipeline_labels.pop(
+                label_weight_column
+            ).to_numpy(dtype=float)
+            dk.feature_pipeline.label_list = pipeline_labels.columns
+            weights = QuickAdapterRegressorV3._enforce_train_weight_support(
```
[the remark this answers is not in the document]
Comment (jerome-benoit) [reply quickadapter/user_data/freqaimodels/QuickAdapterRegressorV3.py:2602]: The regression risk is valid, but committed tests are intentionally not applicable to this PR: its explicit project scope requires focused tests to remain outside the repository, which overrides the generic tests-first preference cited from AGENTS.md. I rechecked the implementation and found no separate code defect.

The published head `cec3f893dc` was revalidated with:

- `/tmp/qa93_repro.py`: a deterministic real DataSieve `Pipeline`/`SVMOutlierExtractor` mask removes two rows after pre-pipeline support passed. It verifies post-filter `raise` for pivot count plus Kish ESS (`1`, `1.04019`), `raise` for positive-label-weight fraction (`1/3`), and `fallback` in the `refit` context. The fallback assertion checks the surviving sanitized base weights exactly as `[0.75, 1.0, 1.25]`.
- The same harness verifies that the pipeline's restored validation label schema is exactly `["&s-extrema"]` and that the no-label-weight path presents only that original label column to DataSieve.
- `/tmp/qa93_runtime.py`: real DataSieve `0.1.9`, scikit-learn `1.6.1`, and `SGDOneClassSVM`; train and validation transforms preserve feature/label/weight shape parity and the original label schema.
- The runtime smoke passed both with Freqtrade `2026.7-dev-02f6ca2d2` and current `develop` source `d9e1cf17adf444cc46bd3bba0f9ff8d38a9b5328`.
- `ruff format --check`, `ruff check quickadapter`, full `compileall`, and `git diff --check` all pass.

No tracked test or code change is warranted under the explicit scope, so I am resolving this thread as non-applicable after documenting the external evidence.

[the remark this answers is not in the document]
Comment (jerome-benoit) [reply quickadapter/user_data/freqaimodels/QuickAdapterRegressorV3.py:2602]: Post-publication zero-audit update for head `87088e836e6e8317754fc387fb3af3b97eb5d7a7`: the request for committed tests remains non-applicable because this task explicitly requires all focused regression harnesses to stay outside the repository. However, the new audit did find and fix one separate code defect: a custom feature transform returning `sample_weight=None` (or a malformed shape) could be silently uniformized and accepted on a one-row training set. The late train-only length check was replaced by one exact shape invariant immediately after `fit_transform`, covering train and refit before sanitization or support fallback.

Fresh external evidence is recorded in the updated PR body: `/tmp/qa93_main_zero.py` reproduces #93 on `upstream/main`; `/tmp/qa93_zero_audit.py` covers `raise`/`fallback`, schema, no-label, NaN/Inf/empty, and `None`/2-D failures against current Freqtrade `develop`; `/tmp/qa93_datasieve_builtin.py` verifies DataSieve row/weight propagation; `/tmp/qa93_five_regressors.py` performs real weighted fits through all five regressors. Ruff, format, Python 3.14 compile, diff checks, and both CodeQL checks pass. No tracked test was added, and the thread remains resolved.

[the remark this answers is not in the document]
Comment (jerome-benoit) [reply quickadapter/user_data/freqaimodels/QuickAdapterRegressorV3.py:2602]: Owner re-audit update for published head `75a24d8920c1bc4fbdad373d0824074e8a17b527`: the committed-test request remains non-applicable under this task's explicit rule that focused regressions stay outside the repository. No tracked test was added.

The exhaustive recheck did reproduce one additional code defect adjacent to the original finding: validation and holdout `feature_pipeline.transform()` could still return `sample_weight=None` for a one-row set and be silently uniformized. The single post-pipeline shape guard now covers train, refit, validation, and holdout before sanitization.

Fresh external evidence is detailed in the updated PR body. In particular, `/tmp/qa150_owner_pipeline_oracle.py` verifies support `raise`/`fallback`, schema restoration, row reordering/index reset, malformed shapes, real DataSieve SVM removal, and fail-closed validation/holdout behavior against Freqtrade `develop` `308a048a`; `/tmp/qa150_owner_regressor_oracle.py` verifies all five real regressors plus HPO, weighted holdout RMSE, and final refit. Ruff, format, compileall, and diff checks pass. The thread remains resolved.
